homeassistant/components/media_player/gpmdp.py

```diff
@@ -0,0 +1,143 @@
+"""
+Support for Google Play Music Desktop Player.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.gpm_dp/
+"""
+import logging
+import json
+import socket
+
+from homeassistant.components.media_player import (
+    MEDIA_TYPE_MUSIC, SUPPORT_NEXT_TRACK, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_PAUSE, MediaPlayerDevice)
+from homeassistant.const import (
+    STATE_PLAYING, STATE_PAUSED, STATE_OFF)
+
+_LOGGER = logging.getLogger(__name__)
+REQUIREMENTS = ['websocket-client==0.35.0']
+SUPPORT_GPMDP = SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the GPMDP platform."""
+    from websocket import create_connection
+
+    name = config.get("name", "GPM Desktop Player")
+    address = config.get("address")
+
+    if address is None:
+        _LOGGER.error("Missing address in config")
+        return False
+
+    add_devices([GPMDP(name, address, create_connection)])
+
+
+class GPMDP(MediaPlayerDevice):
+    """Representation of a GPMDP."""
+
+    # pylint: disable=too-many-public-methods, abstract-method
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, name, address, create_connection):
+        """Initialize."""
+        self._connection = create_connection
+        self._address = address
+        self._name = name
+        self._status = STATE_OFF
+        self._ws = None
+        self._title = None
+        self._artist = None
+        self._albumart = None
+        self.update()
+
+    def get_ws(self):
+        """Check if the websocket is setup and connected."""
+        if self._ws is None:
+            try:
+                self._ws = self._connection(("ws://" + self._address + ":5672"),
+                                            timeout=1)
+            except (socket.timeout, ConnectionRefusedError, ConnectionResetError):
+                self._ws = None
+        elif self._ws.connected is True:
+            self._ws.close()
+            try:
+                self._ws = self._connection(("ws://" + self._address + ":5672"),
+                                            timeout=1)
+            except (socket.timeout, ConnectionRefusedError, ConnectionResetError):
+                self._ws = None
+        return self._ws
+
+    def update(self):
+        """Get the latest details from the player."""
+        websocket = self.get_ws()
+        if websocket is None:
+            self._status = STATE_OFF
+            return
+        else:
+            state = websocket.recv()
+            if ((json.loads(state))['payload']) is True:
+                websocket.recv()
+                websocket.recv()
+                song = self._ws.recv()
```

Comment:
You can make all following lines shorter if you do:

```python
song = json.loads(song)
```

You probably also want to do the same for state so you don't parse it twice.